digdag-server/src/main/java/io/digdag/server/rs/AttemptResource.java

```diff
@@ -344,7 +353,11 @@ private ArchivedTask matchTaskPattern(String pattern, List<ArchivedTask> tasks)
                     tasks
                             .stream()
                             .collect(
-                                    Collectors.toMap(t -> t.getFullName(), t -> t)
+                                    // Avoid 500 error due to duplicate key ArchivedTask
+                                    // if a group retry happened, IllegalStateException will be thrown
+                                    // because there are multiple tasks of the same name.
+                                    // Take the first one to make sure it is appropriately resumed.
+                                    Collectors.toMap(t -> t.getFullName(), t -> t, (a, b) -> a)
```

Comment:
May I ask you the reason you choose first one ?

Reply:
There is no specific reason. Either is fine because of the same name.

Reply:
Ah you need only the name. Got it. Thanks!